augur/filter/include_exclude_rules.py

```diff
@@ -190,8 +184,8 @@ def filter_by_query(metadata, query) -> FilterFunctionReturn:
     set()
 
     """
-    # Try converting all queried columns to numeric.
-    for column in extract_variables(query).intersection(metadata.columns):
+    # Try converting all columns to numeric.
+    for column in metadata.columns:
         metadata[column] = pd.to_numeric(metadata[column], errors='ignore')
```

Comment:
I believe this is converting to numeric in place? Could that potentially affect downstream filters? 
I'm specifically thinking of the `date` column being all year values, so they would convert to numeric. Then they would cause errors in any date filtering queries that expect them to be strings.

Reply:
I think this errors only when using both `--query` and `--exclude-ambiguous-dates-by`. I wrote an [example test](https://github.com/nextstrain/augur/commit/92c2500ae5c8e923ca0a35ab3b8deda726bdc7f1) for this which passes locally on master but fails when rebased onto this branch.

Reply:
Oh yes, thanks for catching this! I added your example test in b5a77e1e0692157b4d657d8956004257fcf83427 and addressed with 9f9bc4f17d963203ca57fcdfcf1545652e37f706.